src/overlays/actors/ovl_En_Dg/z_en_dg.c

```diff
@@ -297,197 +351,232 @@ void func_80989674(EnDg* this, PlayState* play) {
+ * the dog with the Mask of Truth equipped.
+ */
+void EnDg_UpdateTextId(EnDg* this) {
     if (this->index < 14) {
+        // Assuming that the weekEventRegs haven't been tampered with, then this will produce a text ID in
+        // the range of 0x3538 to 0x3545.
         if (this->index % 2) {
             sRacetrackDogInfo[this->index].textId =
-                0x3538 + ((gSaveContext.save.weekEventReg[42 + (this->index / 2)] & (0x10 | 0x20 | 0x40 | 0x80)) >> 4);
+                0x3538 + ((gSaveContext.save.weekEventReg[42 + (this->index / 2)] & 0xF0) >> 4);
         } else {
             sRacetrackDogInfo[this->index].textId =
-                0x3538 + (gSaveContext.save.weekEventReg[42 + (this->index / 2)] & (1 | 2 | 4 | 8));
+                0x3538 + (gSaveContext.save.weekEventReg[42 + (this->index / 2)] & 0x0F);
         }
     } else {
         Actor_MarkForDeath(&this->actor);
     }
 
-    if ((sRacetrackDogInfo[this->index].textId >= 0x3547) || (sRacetrackDogInfo[this->index].textId < 0x3538)) {
+    // As a sanity check, this makes sure the text ID is something in the expected range of 0x3538 to 0x3546.
+    if ((sRacetrackDogInfo[this->index].textId > 0x3546) || (sRacetrackDogInfo[this->index].textId < 0x3538)) {
         sRacetrackDogInfo[this->index].textId = 0x353E;
     }
 
+    // Text ID 0x353D is the text for the Romani Ranch dog, which is in the middle of the race dog block. If
+    // the dog ends up with this text ID, adjust it so the dog says a different message indcating it's in
+    // good condition.
     if (sRacetrackDogInfo[this->index].textId == 0x353D) {
         sRacetrackDogInfo[this->index].textId = 0x3538;
     }
 }
 
-void func_80989D38(EnDg* this, PlayState* play) {
-    if (this->index == 21) {
+void EnDg_StartTextBox(EnDg* this, PlayState* play) {
+    if (this->index == ENDG_INDEX_SWAMP_SPIDER_HOUSE) {
         if (CURRENT_DAY == 1) {
+            // Stupid master...
             Message_StartTextbox(play, 0x91C, NULL);
         } else {
+            // I miss the ranch...
             Message_StartTextbox(play, 0x91E, NULL);
         }
-    } else if ((this->index >= 0) && (this->index < 14)) {
+    } else if ((this->index >= ENDG_INDEX_RACETRACK_0) && (this->index <= ENDG_INDEX_RACETRACK_13)) {
         Message_StartTextbox(play, sRacetrackDogInfo[this->index].textId, NULL);
-    } else if (this->index == 20) {
+    } else if (this->index == ENDG_INDEX_ROMANI_RANCH) {
+        // I wonder if I can stop practicing...
         Message_StartTextbox(play, 0x353D, NULL);
     } else {
+        // Where did that Deku Scrub go?
         Message_StartTextbox(play, 0x627, NULL);
     }
 }
 
-void func_80989E18(EnDg* this, PlayState* play) {
-    RacetrackDogInfo* temp;
-
-    if ((D_8098C2A0 != 0) && !(this->unk_280 & 1)) {
+/**
+ * Checks to see if the player picked up the dog and sets the appropriate state if they did.
+ */
+void EnDg_CheckIfPickedUp(EnDg* this, PlayState* play) {
```

Comment:
`CheckIf` makes me think this is just a function that returns a boolean whether the dog is picked up or not, but this function seems to do more then `Check`. It is hard to think of a good name, but perhaps `EnDg_TryPickUp`?

Reply:
Yeah, I can't think of a good name for this one either. I'll go with your suggestion, but open to any other suggestions too.